jenkins/automation/rgw/ceph/ceph-deploy.groovy

```diff
@@ -0,0 +1,153 @@
+pipeline {
+    agent {
+        node {
+            label 'docker-k8-deployment-node'
+        }
+    }
+
+    options {
+        timeout(time: 240, unit: 'MINUTES')
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '30', numToKeepStr: '30'))
+        ansiColor('xterm')
+        disableConcurrentBuilds()   
+    }
+
+    parameters {
+        string(name: 'CORTX_RE_REPO', defaultValue: 'https://github.com/Seagate/cortx-re/', description: 'Repository for Cluster Setup scripts', trim: true)
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for Cluster Setup scripts', trim: true)
+        text(defaultValue: '''hostname=<hostname>,user=<user>,pass=<password>''', description: 'VM details to be used. First node will be used as Primary node. Please provide a minimum of 3 nodes', name: 'hosts')
+        text(defaultValue: '''/dev/sdX''', description: 'Disks for OSD on VMs. This assumes the disk name and no. of disk is same on all nodes', name: 'OSD_Disks')
```

Comment:
Is there recommendation for minium number of disks

Reply:
yes, we are using single disk on each nodes so for 3 nodes: 3 disks - 3 OSD's
All deployment has been tested with 3 node. It will automatically take new disks as per user input on VMs. It assumes that each vm has same disk with same name as input.